packages/nuxt/src/app/composables/router.ts

```diff
@@ -13,7 +13,7 @@ import type { PageMeta } from '#app'
 import { PageRouteSymbol } from '#app/components/injections'
 
 export const useRouter: typeof _useRouter = () => {
-  return useNuxtApp()?.$router as Router
+  return useNuxtApp()?.$router ?? _useRouter() as Router
```

Comment:
The main issue here is that this will add `useRouter` to the build and not allow it being tree-shaken out if `vue-router` isn't otherwise used.

Reply:
ah ok i did not pay attention that useRouter imported as type only. 
in this case we can lazy loaded like this ?
`const { useRouter } = await import("#vue-router") `

Reply:
unfortunately that would make it asynchronous ...

Reply:
I've run into this issue as well. In a Storybook story, `NuxtLink` will cause this error as it has no access to the Nuxt instance outside of the main application:

```
Error: [nuxt] instance unavailable
  at useRouter (/node_modules/nuxt/dist/app/composables/router.js?v=2fa1c944:8:10))
```

Reply:
I think this should be the responsibility of the storybook module to provide a mocked nuxtApp. It's expected that Nuxt composables expect that nuxtApp is defined and available in their context.